### DISCUSSION TITLE
Update field of fieldArray by array index directly

### DISCUSSION THREAD

**Discussion by zirkelc** [react-hook-form/react-hook-form#5472] [Q&A]
I have a nested component with useFieldArray  and without FormContext (due to performance reasons). I want to update all values of one field of the field array. I don't have access to the setValue method from useForm and would like to avoid to pass it down to the nested component. First idea was to remove and insert the field which needs to be updated. But I also noticed I can simply update the entire field by array index access:

```
      const index = fields.findIndex(f => f.id === field.id);

      fields[index] = {
        id: field.id,
        ...field
      };
```

Is there anything I should keep in mind when doing this, or are there any side effects?

**Comment by bluebill1049** [comment]
can you use `setValue`?

**Reply by zirkelc** [reply]
I would need to pass it down as a prop several levels and would like to avoid it, if I can simply update the field array directly without side  effects?
